### PR TITLE
[terra-dev-site-v7] Consume latest changes from terra-application-integration.

### LINKED ISSUE #326: Upgrade to use terra-functional-testing
# Story

## Description

As a developer of terra I want to upgrade terra-dev-site testing infrastructure to use [terra-functional-testing](https://github.com/cerner/terra-toolkit/tree/main/packages/terra-functional-testing).

## Motivation

So that the project is on the latest version of terra's WDIO 6 testing infrastructure.

## Definition of Done

We'll know we're done when:

- A PR is open to main referencing the snapshot of terra-functional-testing
- All tests are passing

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

This PR pulls in the latest (as of today) changes from terra-application integration including:
* Rendering the 'home' page in a custom layout to avoid the gutter.
* The headless layout no longer renders a page.
* Pages now use the inset gutter.

This PR also updates terra-dev-site to use terra-functional testing.

The directory switcher plugin is now only applied in dev mode.

Default primary nav items now include application dev tools and guides by default.

wdio tests will fail until this PR is released: https://github.com/cerner/terra-toolkit/pull/617

Closes #326 

### Deployment Link

https://terra-dev-site-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: src/layouts/_ContentLayout.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      <ContentLoadedContainer type={pageContentConfig.type} isScrollContainer>
+        <div>Error: chunk failed to load.</div>
+      </ContentLoadedContainer>
+    );
+  }
+
+  return (
+    <MainContainer className={cx('main')}>
+      <Suspense
+        // TODO add a loading spinner?
```
Comment (mjhenkes) [inline src/layouts/_ContentLayout.jsx:46]: Right now there is no loading spinner for a 'main' non page page

**Review by tbiethman — APPROVED** [review]

File: src/content/_ContentLoaded.jsx [hunk 2]
```diff
@@ -42,6 +47,7 @@ const ContentLoaded = ({ children, type }) => {
         cx(
           theme.className,
           ...(['md', 'mdx'].includes(type) ? ['markdown'] : []),
+          ...(isScrollContainer ? ['scroll'] : []),
```
Comment (StephenEsser) [inline src/content/_ContentLoaded.jsx:50]: I think you can use a class conditional here. (probably want to double check this)

```suggestion
          { scroll: isScrollContainer },
```

File: src/content/ContentLoaded.module.scss [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+  .scroll {
+    height: 100%;
+    overflow: auto;
+    -webkit-overflow-scrolling: touch;
+    position: relative;
+    width: 100%;
+  }
+}
+
 :local(.markdown) {
```
Comment (StephenEsser) [inline src/content/ContentLoaded.module.scss:14]: I remember using this syntax before, but I don't remember what this does

File: src/layouts/_ContentLayout.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const ContentLayout = ({ pageContentConfig, contentImports }) => {
+  const location = useLocation();
+  const { pathname } = location;
+  const ContentComponent = contentImports[pathname];
+  const [loadingFailed, setLoadingFailed] = React.useState();
+
+  if (!pageContentConfig) {
+    return <div>404</div>;
```
[the remark this answers is not in the document]
Comment (mjhenkes) [reply src/layouts/_ContentLayout.jsx:32]: ![https://media.giphy.com/media/NEvPzZ8bd1V4Y/giphy.gif]

File: wdio.conf.js [hunk 5]
```diff
@@ -1,12 +1,14 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
-const defaultWdioConfig = require('terra-toolkit/config/wdio/wdio.conf');
+const { config } = require('@cerner/terra-functional-testing/lib/config/wdio.conf');
```
Comment (StephenEsser) [inline wdio.conf.js:2]:
```suggestion
const { config } = require('@cerner/terra-functional-testing');
```

**Review by StephenEsser — APPROVED** [review]

File: src/content/ContentLoaded.module.scss (hunk 3 above)
Comment (mjhenkes) [reply src/content/ContentLoaded.module.scss:14]: That class specifically is local (hashed), but everything under it is global (not hashed).

**Review by ryanthemanuel — APPROVED** [review]

File: src/content/_ContentLoaded.jsx (hunk 2 above)
Comment (mjhenkes) [reply src/content/_ContentLoaded.jsx:50]: That does totally work. Classnames has a pretty flexible api.

File: src/layouts/_ContentLayout.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mjhenkes) [reply src/layouts/_ContentLayout.jsx:32]: I'll make this update in a future PR.

**Pull request merged**